src/base/IoManip.h

```diff
@@ -75,22 +105,61 @@ operator <<(std::ostream &os, const RawPointerT<Pointer> &pd)
     return os;
 }
 
-/// std::ostream manipulator to print integers as hex numbers prefixed by 0x
+/// std::ostream manipulator to print integers and alike as hex numbers.
+/// Normally used through the asHex() convenience function.
 template <class Integer>
 class AsHex
 {
 public:
+    // Without this assertion, asHex(pointer) and AsHex(3.14) compile, but their
+    // caller is likely confused about the actual argument type and expects
+    // different output. Enum values are not integral types but arguably do not
+    // cause similar problems.
+    static_assert(std::is_integral<Integer>::value || std::is_enum<Integer>::value);
+
     explicit AsHex(const Integer n): io_manip(n) {}
+
+    /// Sets the minimum number of digits to print. If the integer has fewer
+    /// digits than the given width, then we also print leading zero(s).
+    /// Otherwise, this method has no effect.
+    auto &minDigits(const size_t w) { forcePadding = w; return *this; }
+
+    /// Print hex digits in upper (or, with a false parameter value, lower) case.
+    auto &upperCase(const bool u = true) { forceCase = u; return *this; }
+
     Integer io_manip; ///< the integer to print
+
+    ///< \copydoc minDigits(). The default is to use stream's field width and
+    ///stream's fill character.
```

Comment:
The "///<" indicates that this line is documentation for the previous (`io_manip`) member. Looks like a typo to me.
Also, it seems best to have `\copydoc` on its own line since that will pull in a whole set of self-contained documentation syntax.
```suggestion
    /// \copydoc minDigits()
    /// The default is to use stream's field width and stream's fill character.
```